blaze-core/src/main/scala/org/http4s/blazecore/util/package.scala

```diff
@@ -29,23 +27,7 @@ package object util {
   private[http4s] val FutureUnit =
     Future.successful(())
 
-  // Adapted from https://github.com/typelevel/cats-effect/issues/199#issuecomment-401273282
-  /** Inferior to `Async[F].fromFuture` for general use because it doesn't shift, but
-    * in blaze internals, we don't want to shift.
-    */
   private[http4s] def fromFutureNoShift[F[_], A](f: F[Future[A]])(implicit F: Async[F]): F[A] =
-    F.flatMap(f) { future =>
-      future.value match {
-        case Some(value) =>
-          F.fromTry(value)
-        case None =>
-          F.async { cb =>
-            future.onComplete {
-              case Success(a) => cb(Right(a))
-              case Failure(t) => cb(Left(t))
-            }(directec)
-          }
-      }
-    }
+    F.fromFuture(f)
```

Comment:
I don't know if we need a special function for this now.

Reply:
We do, the comment  in the old diff is still relevant, you don't want to shift back onto the IO pool

Reply:
https://github.com/http4s/http4s/pull/3894/commits/aab6fbbed0d9554453d537a3f6be906818579aa3